pkgs/development/libraries/ncurses/default.nix

```diff
@@ -43,7 +43,17 @@ stdenv.mkDerivation rec {
     ++ lib.optionals stdenv.hostPlatform.isWindows [
       "--enable-sp-funcs"
       "--enable-term-driver"
-    ];
+  ] ++ lib.optionals (stdenv.hostPlatform.isUnix && stdenv.hostPlatform.isStatic) [
+      # For static binaries, the point is to have a standalone binary with
+      # minimum dependencies. So here we make sure that binaries using this
+      # package won't depend on a terminfo database located in the Nix store.
+      "--with-terminfo-dirs=${lib.concatStringsSep ":" [
```

Comment:
It would be nice to put the store path to the terminfo database first. This way on the original nixos system ncurses still works purely. Having some global paths as default might even be a good idea on shared builds.

Reply:
It looks like putting the store path at the beginning of the default search path won't actually change the behavior on NixOS.

On NixOS systems, the store path is searched last because NixOS sets the `TERMINFO_DIRS` environment variable to make ncurses search for the Nix profile paths first. This way, it becomes possible for users to make system-wide changes to the terminfo database.

https://github.com/NixOS/nixpkgs/blob/4f390b19e90fde4f729d52f887ab4410a104d6c3/nixos/modules/config/terminfo.nix#L39-L41

Looking at the man page for terminfo, the store path will still be used as a fallback path after this change:

> Fetching Compiled Descriptions
>     The ncurses library searches for terminal descriptions in several places.  It uses only the first description found.  The
>     library  has a compiled-in list of places to search which can be overridden by environment variables.  Before starting to
>     search, ncurses eliminates duplicates in its search list.
>
>     •   If the environment variable TERMINFO is set, it is interpreted as the pathname of a directory containing the compiled
>         description you are working on.  Only that directory is searched.
>
>     •   If TERMINFO is not set, ncurses will instead look in the directory $HOME/.terminfo for a compiled description.
>
>     •   Next,  if  the  environment  variable TERMINFO_DIRS is set, ncurses will interpret the contents of that variable as a
>         list of colon-separated directories (or database files) to be searched.
>
>         An empty directory name (i.e., if the variable begins or ends with a colon, or contains adjacent  colons)  is  inter‐
>         preted as the system location /nix/store/s1jg1iav1d9wgvcikybj2xizaj52a9ga-ncurses-6.2/share/terminfo.
>
>     •   Finally, ncurses searches these compiled-in locations:
>
>         •   a list of directories (no default value), and
>
>         •   the  system  terminfo directory, /nix/store/s1jg1iav1d9wgvcikybj2xizaj52a9ga-ncurses-6.2/share/terminfo (the com‐
>             piled-in default).

I also corrected the original PR description, because I realized that static binaries would work across different NixOS systems as long as `TERMINFO_DIRS` is left unchanged.

Reply:
> Having some global paths as fallback might even be a good idea on shared builds.

I guess this is for non-NixOS systems?

Reply:
If the system terminfo directory is always searched then you can disregard my comment.